### PR TITLE
port etcd confchange test

### PR DESCRIPTION
**Pull request by accelsao**
Signed-off-by: accelsao <jayzhan211@gmail.com>

### CODE REVIEW

**Comment by accelsao** [comment]
A problem I meet currently, function with Copy expect mutable reference type to be implemented Copy.

**Comment by accelsao** [comment]
@BusyJay, A new PR, because I don't know how to use git ...

**Comment by accelsao** [comment]
Is there a reason that 
`apply_conf` in `ProgressTracker` is not written in `Changer` ?

**Comment by accelsao** [comment]
Previous problems are solved

**Comment by accelsao** [comment]
@BusyJay , Ready for Review

File: src/tracker.rs [hunk 1]
```diff
@@ -88,6 +89,48 @@ pub struct Configuration {
...
 }
 
+// Display and crate::itertools used only for test
+#[cfg(test)]
+impl std::fmt::Display for Configuration {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use itertools::Itertools;
+        let mut buffer = String::new();
+        if self.voters.outgoing.is_empty() {
+            buffer.push_str(&format!("voters={}", self.voters.incoming));
```
Comment (BusyJay) [inline src/tracker.rs:99]: Why not just `write!(f, "voters={}", self.voters.incoming)?`?

File: src/confchange/datadriven_test.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                            }
+                            Err(e) => {
+                                idx += 1;
+                                return e.to_string();
+                            }
+                        }
+                    }
+                    "leave-joint" => {
+                        assert!(data.cmd_args.is_empty());
+                        match Changer::new(&tr).leave_joint() {
```
Comment (BusyJay) [inline src/confchange/datadriven_test.rs:59]: The match is so common that it can be extracted to L74.

**Review by BusyJay — COMMENTED** [review]
Rest LGTM

**Comment by accelsao** [comment]
@BusyJay PTAL

**Review by BusyJay — APPROVED** [review]
Thanks! @NingLin-P PTAL

**Review by Fullstop000 — APPROVED** [review]

**Comment by BusyJay** [comment]
Thanks @accelsao @Fullstop000

**Pull request merged**